sbpack/noncwl/utils.py

```diff
@@ -155,6 +188,80 @@ def get_readme(path):
     return None
 
 
+def get_tower_yml(path):
+    """
+    Find tower.yml file is there is one in the path folder
+    """
+    for file in os.listdir(path):
+        if file.lower() == 'tower.yml':
+            return os.path.join(path, file)
+    return None
+
+
+def get_entrypoint(path):
+    """
+    Auto find main.nf or similar file is there is one in the path folder.
+    """
+    possible_paths = []
+    for file in os.listdir(path):
+        if file.lower() == 'main.nf':
+            return file
+
+        if file.lower().endswith('.nf'):
+            possible_paths.append(file)
+
+    if possible_paths:
+        return possible_paths.pop()
+    return None
+
+
+def get_config_files(path):
+    """
+    Auto find config files.
+    """
+    paths = []
+    for file in os.listdir(path):
+        if file.lower().endswith('.config'):
+            paths.append(os.path.join(path, file))
+    return paths or None
+
+
+def parse_config_file(file_path):
+    profiles_text = ""
+
+    with open(file_path, 'r') as file:
+        found_profiles = False
+        brackets = 0
+
+        for line in file.readlines():
+            if found_profiles:
+                profiles_text += line
+                brackets += line.count("{") - line.count("}")
+
+            if brackets < 0:
+                break
+
+            if re.findall(r'profiles\s+\{', line):
```

Comment:
There must be a better and a more reliable way than to parse Nextflow's modular configuration files with regular expressions...

To start with, we could execute say a `nextflow config -a -flat` command to let nextflow itself resolve all the includes and output a format that is easier to parse. For example:

```
[Fri Jul 28 14:38:32 2023] (nf) ~/nextflow/tests  master
@ nextflow -C profiles.config config -a -flat
echo = true
foo = 'bar'
profiles.standard.process.cpus = 2
profiles.standard.process.memory = '2GB'
profiles.advanced.process.cpus = 8
profiles.advanced.process.memory = '10GB'
profiles.advanced.process.disk = '300GB'
```

Reply:
I've tried this before, but it did not match what was expected. If I do nf_core.utils.nextflow_cmd(f"nextflow config -a -flat /path/to/viralrecon-master") (repo: https://github.com/nf-core/viralrecon) i get:

- abims
- adcra
- aither
- alice
- apate
- archgen
- arges
- arm
- aws_tower
- awsbatch
- azurebatch
- bi
- bigpurple
- binac
- biohpc_gen
- biowulf
- c16r64
- c2r8
- c4r16
- c8r32
- cambridge
- cbe
- ccga_dx
- ccga_med
- cedars
- ceres
- cfc
- cfc_dev
- charliecloud
- cheaha
- cobra
- computerome
- conda
- cpu_milan
- cpu_rome
- cpu_rome_512
- cpu_rome_all
- crg
- crick
- crukmi
- czbiohub_aws
- debug
- debug_rome
- denbi_qbic
- dev_prio
- devel
- dkfz
- docker
- doduo
- ebc
- ebi_codon
- ebi_codon_slurm
- eddie
- engaging
- eva
- fgcz
- fub_curta
- genius
- genotoul
- genouest
- gis
- gitpod
- google
- googlebatch
- googlels
- gpu_rome_a100
- gpu_rome_a100_40
- gpu_rome_a100_80
- hasta
- highpriority
- hki
- hypatia
- icr_davros
- ifb_core
- imperial
- incliva
- ipop_up
- janelia
- jax
- kirlia
- ku_sund_dangpu
- leicester
- long
- lugh
- maestro
- mamba
- mana
- marvin
- medair
- medbio
- mem
- mjolnir_globe
- mpcdf
- munin
- normal
- nu_genomics
- oist
- pasteur
- pawsey_nimbus
- phoenix
- podman
- prince
- prod_prio
- production
- psmn
- raven
- rosalind
- rosalind_uge
- sage
- sahmri
- sanger
- sbc_sharc
- scw
- seawulf
- seg_globe
- shifter
- singularity
- skitty
- stub
- stub_prio
- superdome
- swalot
- test
- test_full
- test_full_illumina
- test_full_nanopore
- test_full_sispa
- test_nanopore
- test_sispa
- tigem
- tubingen_apg
- ucl_myriad
- uct_hpc
- uge
- unibe_ibu
- uppmax
- utd_ganymede
- utd_sysbio
- uw_hyak_pedslabs
- uzh
- vai
- victini
- vsc_kul_uhasselt
- vsc_ugent
- wehi
- wgs
- wice
- wustl_htcf
- xanadu

but I expect:

- arm
- charliecloud
- conda
- debug
- docker
- gitpod
- mamba
- podman
- shifter
- singularity
- test
- test_full
- test_full_illumina
- test_full_nanopore
- test_full_sispa
- test_nanopore
- test_sispa

Additionally, one of the profiles (arm), was in a warning message instead of being listed properly:
WARN: Unknown config attribute `profiles.arm` -- check config file: /Users/pavlemarinkovic/Downloads/viralrecon-master/nextflow.config

There might be a better way to parse the config file, but I've not found it. From what I gather, nextflow is using a custom format to write config files, so I cannot load them in as with JSON or YAML.

Reply:
Ok, we'll go with this approach for now.